pkg/schemas/task_parameters.go

```diff
@@ -0,0 +1,91 @@
+		sdk.TaskParameterTaskAutoRetryAttempts,
+		sdk.TaskParameterUserTaskManagedInitialWarehouseSize,
+		sdk.TaskParameterUserTaskMinimumTriggerIntervalInSeconds,
+		sdk.TaskParameterUserTaskTimeoutMs,
+		// session parameters
+		sdk.TaskParameterAbortDetachedQuery,
+		sdk.TaskParameterAutocommit,
+		sdk.TaskParameterBinaryInputFormat,
+		sdk.TaskParameterBinaryOutputFormat,
+		sdk.TaskParameterClientMemoryLimit,
+		sdk.TaskParameterClientMetadataRequestUseConnectionCtx,
+		sdk.TaskParameterClientPrefetchThreads,
+		sdk.TaskParameterClientResultChunkSize,
+		sdk.TaskParameterClientResultColumnCaseInsensitive,
+		sdk.TaskParameterClientSessionKeepAlive,
+		sdk.TaskParameterClientSessionKeepAliveHeartbeatFrequency,
+		sdk.TaskParameterClientTimestampTypeMapping,
+		sdk.TaskParameterDateInputFormat,
+		sdk.TaskParameterDateOutputFormat,
+		sdk.TaskParameterEnableUnloadPhysicalTypeOptimization,
+		sdk.TaskParameterErrorOnNondeterministicMerge,
+		sdk.TaskParameterErrorOnNondeterministicUpdate,
+		sdk.TaskParameterGeographyOutputFormat,
+		sdk.TaskParameterGeometryOutputFormat,
+		sdk.TaskParameterJdbcTreatTimestampNtzAsUtc,
+		sdk.TaskParameterJdbcUseSessionTimezone,
+		sdk.TaskParameterJsonIndent,
+		sdk.TaskParameterLockTimeout,
+		sdk.TaskParameterLogLevel,
+		sdk.TaskParameterMultiStatementCount,
+		sdk.TaskParameterNoorderSequenceAsDefault,
+		sdk.TaskParameterOdbcTreatDecimalAsInt,
+		sdk.TaskParameterQueryTag,
+		sdk.TaskParameterQuotedIdentifiersIgnoreCase,
+		sdk.TaskParameterRowsPerResultset,
+		sdk.TaskParameterS3StageVpceDnsName,
+		sdk.TaskParameterSearchPath,
+		sdk.TaskParameterStatementQueuedTimeoutInSeconds,
+		sdk.TaskParameterStatementTimeoutInSeconds,
+		sdk.TaskParameterStrictJsonOutput,
+		sdk.TaskParameterTimestampDayIsAlways24h,
+		sdk.TaskParameterTimestampInputFormat,
+		sdk.TaskParameterTimestampLtzOutputFormat,
+		sdk.TaskParameterTimestampNtzOutputFormat,
+		sdk.TaskParameterTimestampOutputFormat,
+		sdk.TaskParameterTimestampTypeMapping,
+		sdk.TaskParameterTimestampTzOutputFormat,
+		sdk.TaskParameterTimezone,
+		sdk.TaskParameterTimeInputFormat,
+		sdk.TaskParameterTimeOutputFormat,
+		sdk.TaskParameterTraceLevel,
+		sdk.TaskParameterTransactionAbortOnError,
+		sdk.TaskParameterTransactionDefaultIsolationLevel,
+		sdk.TaskParameterTwoDigitCenturyStart,
+		sdk.TaskParameterUnsupportedDdlAction,
+		sdk.TaskParameterUseCachedResult,
+		sdk.TaskParameterWeekOfYearPolicy,
+		sdk.TaskParameterWeekStart,
+	}
+)
+
+func init() {
+	for _, param := range taskParameters {
+		ShowTaskParametersSchema[strings.ToLower(string(param))] = ParameterListSchema
+	}
+}
+
+func TaskParametersToSchema(parameters []*sdk.Parameter) map[string]any {
+	taskParametersValue := make(map[string]any)
+	for _, param := range parameters {
+		if slices.Contains(taskParameters, sdk.TaskParameter(param.Key)) {
+			taskParametersValue[strings.ToLower(param.Key)] = []map[string]any{ParameterToSchema(param)}
+		}
```

Comment:
I think we should log, if the given param is not in the list of taskParameters.

Reply:
Yeah, but currently, we don't do it for any other parameters schema, and I think even if we log it, it wouldn't really see it until someone gets an error because of it. Plus, there are a lot of parameters that are in the tasks that are not handled here. That would mean for every read for every task we would log at least 60 parameters (same for users).